activation/activation_test.go

```diff
@@ -366,6 +369,7 @@ func TestBuilder_StartSmeshingAvoidsPanickingIfPrepareInitializerReturnsError(t
 	goroutineCount := runtime.NumGoroutine()
 	tab.StartSmeshing(tab.coinbase, PostSetupOpts{})
 	require.Equal(t, goroutineCount, runtime.NumGoroutine())
+	tab.eg.Wait()
```

Comment:
So I think this would solve the issue with getting stuck, it is quite tricky though. Given the trickyness of this test, I'd also be happy with just removing the test altogether.
```suggestion
	closeChan := make(chan struct{})
	go func() {
		tab.eg.Wait()
		closeChan <- struct{}{}
	}()
	select {
	case closeChan <- struct{}{}:
	case <-time.NewTimer(time.Second * 5).C:
		tab.stop()
		<-closeChan
		t.Fatal("timed out waiting for close")
	}
```

Reply:
This kinda defeats the whole purpose of the `errgroup`: If the timer fires then the test is aborted with a go routine still running in the background when the next test starts.

Imo we should just let the test get stuck, we have timeouts in place that abort test execution if a test times out: `go test -timeout 5m` so if a single test takes longer than 5 mins to complete test execution will be aborted, which imo is a safer approach.

Reply:
It is waiting for the go-routines to finish, after the timeout it waits on the close channel for the errgroup to return.

But it is tricky! Which is probably why you misread it.